### PR TITLE
Add price per item to featured product

### PR DESCRIPTION
**Pull request by eugenekasimov**
### PR Summary: 

Add price per item change to the feature product section

Editor: https://os2-demo.myshopify.com/admin/themes/140247466006/editor

### CODE REVIEW

File: sections/featured-product.liquid [hunk 1]
```diff
@@ -22,6 +22,8 @@
 {%- endstyle -%}
 
 <script src="{{ 'product-info.js' | asset_url }}" defer="defer"></script>
+<script src="{{ 'show-more.js' | asset_url }}" defer="defer"></script>
+<script src="{{ 'price-per-item.js' | asset_url }}" defer="defer"></script>
```
Comment (sofiamatulis) [inline sections/featured-product.liquid:26]:
```suggestion
{%- if product.quantity_price_breaks_configured? -%}
   <script src="{{ 'show-more.js' | asset_url }}" defer="defer"></script>
  <script src="{{ 'price-per-item.js' | asset_url }}" defer="defer"></script>
{%- endif -%}
```

**Review by sofiamatulis — APPROVED** [review]
Works well 👍

File: sections/featured-product.liquid [hunk 2]
```diff
@@ -242,36 +246,71 @@
...
+                      <button class="quantity__button no-js-hidden" name="plus" type="button">
+                        <span class="visually-hidden">
+                          {{- 'products.product.quantity.increase' | t: product: product.title | escape -}}
+                        </span>
+                        {% render 'icon-plus' %}
+                      </button>
+                    </quantity-input>
+                    {%- liquid
+                      assign volume_pricing_array = product.selected_or_first_available_variant.quantity_price_breaks | sort: 'quantity' | reverse
+                      assign current_qty_for_volume_pricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
```
Comment (KaichenWang) [inline sections/featured-product.liquid:282]: I think using `min` here can cause incorrect price per item for products with Quantity rules configured. We should also fix this for the PDP (`main-product.liquid`):
Screenshot 2023-07-10 at 7 29 43 PM png2

```suggestion
                      assign current_qty_for_volume_pricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.increment
```

Comment (sofiamatulis) [reply sections/featured-product.liquid:282]: I think we need to do that only when there are items in cart

```assign current_qty_for_volume_pricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.min
if cart_qty > 0
  assign current_qty_for_volume_pricing = cart_qty | plus: product.selected_or_first_available_variant.quantity_rule.increment
endif```

Comment (sofiamatulis) [reply sections/featured-product.liquid:282]: https://github.com/Shopify/dawn/pull/2799/commits/7e44881694621b85af14dd73b7ef483716bf2723

**Review by KaichenWang — APPROVED** [review]
Tested the updates and looks good! We can continue testing other scenarios together in the base branch.

**Pull request merged**